python/dgl/convert.py

```diff
@@ -785,89 +744,327 @@ def to_homo(G):
+       :class:`scipy.sparse.csr.csr_matrix`). It converts the input matrix to the COOrdinate
+       format using :func:`scipy.sparse.spmatrix.tocoo` before creates a :class:`DGLGraph`.
+       Creating from a :class:`scipy.sparse.coo.coo_matrix` is hence the most efficient way.
+    2. DGL internally maintains multiple copies of the graph structure in different sparse
+       formats and chooses the most efficient one depending on the computation invoked.
+       If memory usage becomes an issue in the case of large graphs, use
+       :func:`dgl.DGLGraph.formats` to restrict the allowed formats.
+
+    Examples
+    --------
+
+    The following example uses PyTorch backend.
+
+    >>> import dgl
+    >>> import numpy as np
+    >>> import torch
+    >>> from scipy.sparse import coo_matrix
+
+    Create a small three-edge graph.
+
+    >>> # Source nodes for edges (2, 1), (3, 2), (4, 3)
+    >>> src_ids = np.array([2, 3, 4])
+    >>> # Destination nodes for edges (2, 1), (3, 2), (4, 3)
+    >>> dst_ids = np.array([1, 2, 3])
+    >>> # Weight for edges (2, 1), (3, 2), (4, 3)
+    >>> eweight = np.array([0.2, 0.3, 0.5])
+    >>> sp_mat = coo_matrix((eweight, (src_ids, dst_ids)), shape=(5, 5))
+    >>> g = dgl.from_scipy(sp_mat)
+
+    Retrieve the edge weights.
+
+    >>> g = dgl.from_scipy(sp_mat, eweight_name='w')
+    >>> g.edata['w']
+    tensor([0.2000, 0.3000, 0.5000], dtype=torch.float64)
+
+    Create a graph on the first GPU card with data type int32.
+
+    >>> g = dgl.from_scipy(sp_mat, idtype=torch.int32, device='cuda:0')
+
+    See Also
+    --------
+    graph
+    from_networkx
     """
+    # Sanity check
+    utils.check_type(sp_mat, spmatrix, 'sp_mat', skip_none=False)
+    num_rows = sp_mat.shape[0]
+    num_cols = sp_mat.shape[1]
+    if num_rows != num_cols:
+        raise DGLError('Expect the number of rows to be the same as the number of columns for '
+                       'sp_mat, got {:d} and {:d}.'.format(num_rows, num_cols))
+    utils.check_valid_idtype(idtype)
+
     u, v, urange, vrange = utils.graphdata2tensors(sp_mat, idtype)
-    g = create_from_edges(u, v, ntype, etype, ntype, urange, vrange,
-                          validate=False, formats=formats)
+    g = create_from_edges(u, v, '_N', '_E', '_N', urange, vrange)
     if eweight_name is not None:
         g.edata[eweight_name] = F.tensor(sp_mat.data)
-    return g
+    return g.to(device)
+
+def bipartite_from_scipy(sp_mat,
+                         utype='_U', etype='_E', vtype='_V',
+                         eweight_name=None,
+                         idtype=None,
+                         device=None):
+    """Create a unidirectional bipartite graph from a SciPy sparse matrix.
+
+    A bipartite graph has two types of nodes ``"SRC"`` and ``"DST"`` and
+    there are only edges between nodes of different types. By "unidirectional",
+    there are only edges from ``"SRC"`` nodes to ``"DST"`` nodes.
```

Comment:
Need to rewrite this part.  Here is a try:
```suggestion
    The created graph will have two types of nodes ``utype`` and ``vtype``, as
    well as one edge type ``etype`` whose edges are from ``utype`` to ``vtype``.
```

Reply:
In such case, do we still need to mention "unidirectional bipartite graph" in the beginning of the docstring?

Reply:
Done.